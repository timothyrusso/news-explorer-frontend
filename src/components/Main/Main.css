.main {
  width: 100%;
  min-height: 624px;
}

.about {
  display: flex;
  justify-content: space-between;
  padding: 80px 216px 80px 104px;
}

.about__image {
  max-width: 464px;
  border-radius: 1000px;
}

.about__wrapper {
  max-width: 600px;
}

.about__title {
  margin: 0;
  padding: 51px 0 24px;
  font-family: 'Roboto Slab';
  font-style: normal;
  font-weight: 400;
  font-size: 40px;
  line-height: 1.15;
}

.about__description {
  margin: 0;
  padding-bottom: 25px;
  font-family: 'Roboto';
  font-style: normal;
  font-weight: 400;
  font-size: 18px;
  line-height: 1.33;
}

.about__description:last-child {
  padding-bottom: 0;
}

@media screen and (max-width: 1400px) {
  .main {
    min-height: 424px;
  }

  .about__image {
    max-width: 350px;
  }
}

@media screen and (max-width: 1300px) {
  .about__image {
    max-width: 260px;
  }
}

@media screen and (max-width: 1200px) {
  .about__wrapper {
    margin-left: 30px;
  }
}

@media screen and (max-width: 1200px) {
  .about__image {
    height: 260px;
  }
}

@media screen and (max-width: 930px) {
  .main {
    min-height: 354px;
  }

  .about {
    padding: 40px 40px 61px;
    max-width: 700px;
  }

  .about__image {
    max-width: 232px;
    height: 232px;
    margin-top: 21px;
    box-sizing: border-box;
  }

  .about__wrapper {
    max-width: 424px;
  }

  .about__title {
    font-size: 30px;
    line-height: 1.33;
    padding: 0 0 16px;
  }

}

@media screen and (max-width: 650px) {
  .main {
    min-height: 752px;
  }

  .about {
    display: flex;
    flex-direction: column;
    padding: 32px 16px 128px;
  }

  .about__wrapper {
    display: flex;
    flex-direction: column;
  }

  .about__image {
    max-width: 272px;
    height: 272px;
    margin: 0 8px 24px;
  }
}
