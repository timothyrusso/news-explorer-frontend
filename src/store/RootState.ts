import { UserState } from './user/user.reducer';
import { ArticleState } from './article/article.reducer';
import { TogglesState } from './toggles/toggles.reducer';
import { ErrorsState } from './errors/errors.reducer';

export type RootState = {
  user: UserState;
  article: ArticleState;
  toggles: TogglesState;
  errors: ErrorsState;
};
