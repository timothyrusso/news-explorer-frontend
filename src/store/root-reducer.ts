import { combineReducers } from 'redux';
import { userReducer } from './user/user.reducer';
import { articleReducer } from './article/article.reducer';

export const rootReducer = combineReducers({
  user: userReducer,
  article: articleReducer,
});
