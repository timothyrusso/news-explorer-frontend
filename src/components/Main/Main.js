import "./Main.css";
import aboutImage from "../../images/about_me_image.jpg";

const Main = () => {
  return (
    <main className="main">
      <section className="about">
        <img
          src={aboutImage}
          className="about__image"
          alt="A smartphone that is showing a dashboard"
        ></img>
        <div className="about__wrapper">
          <h2 className="about__title">About the author</h2>
          <p className="about__description">
            Hey there, I'm Timothy. I'm a motivated and passionate 
            software engineer with a background in eCommerce, Quality Assurance, Application
            Delivery Management, Project Management and Trade Marketing. I have
            a bachelor's degree in Economics and Business Management and I have done a coding bootcamp called Practicum.
          </p>
          <p className="about__description">
            Iâ€™m a curious person, I love to code and I like to experiment with
            new technologies. Now coding is my passion and my professional
            career focus.
          </p>
        </div>
      </section>
    </main>
  );
};

export default Main;
