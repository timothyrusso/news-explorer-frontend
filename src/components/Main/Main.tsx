import React from 'react';
import aboutImage from '../../images/about_me_image.jpg';
import './Main.css';

const Main = () => {
  return (
    <main className="main">
      <section className="about">
        <img
          src={aboutImage}
          className="about__image"
          alt="Timothy - a passionate frontend engineer"
        ></img>
        <article className="about__wrapper">
          <h2 className="about__title">About the author</h2>
          <p className="about__description">
            Hey there, I'm Timothy. I'm a motivated and passionate frontend
            engineer with a background in eCommerce, Quality Assurance,
            Application Delivery Management, Project Management and Trade
            Marketing. I have a bachelor's degree in Economics and Business
            Management and I have done a coding bootcamp called TripleTen.
          </p>
          <p className="about__description">
            Iâ€™m a curious person, I love to code and I like to experiment with
            new technologies. Now coding is my passion and my professional
            career focus.
          </p>
        </article>
      </section>
    </main>
  );
};

export default Main;
