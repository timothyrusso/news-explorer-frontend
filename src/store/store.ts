import { legacy_createStore as createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { rootReducer } from './root-reducer';
import { ThunkDispatch } from 'redux-thunk';
import { AnyAction } from 'redux';
import { RootState } from './RootState';

const store = createStore(rootReducer, applyMiddleware(thunk));

export type AppDispatch = ThunkDispatch<RootState, unknown, AnyAction>;

export default store;
