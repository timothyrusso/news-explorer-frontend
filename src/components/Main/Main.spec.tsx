import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../store/store';
import Main from './Main';

describe('Main component', () => {
  it('Should render the Main component', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Main />
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
