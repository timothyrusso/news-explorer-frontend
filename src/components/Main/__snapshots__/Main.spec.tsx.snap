// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component Should render the Main component 1`] = `
<DocumentFragment>
  <main
    class="main"
  >
    <section
      class="about"
    >
      <img
        alt="Timothy - a passionate software engineer"
        class="about__image"
        src="test-file-stub"
      />
      <article
        class="about__wrapper"
      >
        <h2
          class="about__title"
        >
          About the author
        </h2>
        <p
          class="about__description"
        >
          Hey there, I'm Timothy. I'm a motivated and passionate software engineer with a background in eCommerce, Quality Assurance, Application Delivery Management, Project Management and Trade Marketing. I have a bachelor's degree in Economics and Business Management and I have done a coding bootcamp called Practicum.
        </p>
        <p
          class="about__description"
        >
          Iâ€™m a curious person, I love to code and I like to experiment with new technologies. Now coding is my passion and my professional career focus.
        </p>
      </article>
    </section>
  </main>
</DocumentFragment>
`;
